ui/hooks/use-user.ts

```diff
@@ -0,0 +1,28 @@
+import { create } from "zustand";
+import { createJSONStorage, persist } from "zustand/middleware";
+
+import { UserProfileProps } from "@/types";
+
+type UserStore = {
+  user: UserProfileProps | null;
+  setUser: (user: UserProfileProps | null) => void;
+  clearUser: () => void;
+};
+
+export const useUser = create(
```

Comment:
Not sure this is the right approach. This stores user info in localStorage, which raises security concerns especially when we already have `useSession` available.

If we need shared access to user data, we could fetch it once in the layout and pass it via context, without persisting anything in the browser.